src/binary.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+import {Tensor} from './tensor.js';
+import {getBroadcastShape, product} from './common.js';
+
+/**
+ *  Compute binary operation of the two input numbers and given operation name.
+ * @param {Number} a
+ * @param {Number} b
+ * @param {String} name 'add'|'sub'|'mul'|'div'|'max'|'min'|'pow'
+ * @return {Number}
+ */
+function computeBinary(a, b, name) {
+  let out;
+  switch (name) {
+    case 'add':
+      out = a + b;
+      break;
+    case 'sub':
+      out = a - b;
+      break;
+    case 'mul':
+      out = a * b;
+      break;
+    case 'div':
+      out = a / b;
+      break;
+    case 'max':
+      out = Math.max(a, b);
+      break;
+    case 'min':
+      out = Math.min(a, b);
+      break;
+    case 'pow':
+      out = Math.pow(a, b);
+      break;
+    default:
+      throw new Error(`'${name}' is not supported`);
+  }
+  return out;
+}
+
+/**
+ * Compute the element-wise binary operation of the two input tensors and given operation name.
+ * @param {Tensor} inputA
+ * @param {Tensor} inputB
+ * @param {String} name 'add'|'sub'|'mul'|'div'|'max'|'min'|'pow'
+ * @return {Tensor}
+ */
+export function binary(inputA, inputB, name) {
```

Comment:
The third parameter probably could be a function object that computes the binary.

Reply:
I prefer to having individual function for each binary op, such as `add`, `sub`. You can just call the `binary` with a corresponding function object.

Reply:
Thanks your suggestion, I will update it.